lightning-invoice/src/utils.rs

```diff
@@ -430,12 +460,22 @@ fn filter_channels(channels: Vec<ChannelDetails>, min_inbound_capacity_msat: Opt
 	// If all channels are private, return the route hint for the highest inbound capacity channel
 	// per counterparty node. If channels with an higher inbound capacity than the
 	// min_inbound_capacity exists, filter out the channels with a lower capacity than that.
-	filtered_channels.into_iter()
-		.filter(|(_counterparty_id, channel)| {
-			min_capacity_channel_exists && channel.inbound_capacity_msat >= min_inbound_capacity ||
-			!min_capacity_channel_exists
+	filtered_channels
+		.into_iter()
+		.map(|(_, channel)| channel)
+		.filter(|channel| {
+			let has_enough_inbound_capacity = channel.inbound_capacity_msat >= min_inbound_capacity;
+			let include_channel = !min_capacity_channel_exists || has_enough_inbound_capacity;
+			if !include_channel {
+				log_debug!(logger, "Ignoring channel {} without enough capacity for invoice route hints",
+					log_bytes!(channel.channel_id));
+			}
+			include_channel
```

Comment:
This will clash a little with the cleanup in #1769 :). I have no preference for which version we go with, but could be worth considering by either version.